src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -344,16 +344,43 @@ class IpexCommunicationService extends AgentService {
   }
 
   async createLinkedIpexMessageRecord(
-    credentialType: string,
-    connectionId: string,
     message: IpexMessage,
     historyType: ConnectionHistoryType
   ): Promise<void> {
+    const schemaSaid = message.exn.e.acdc.s;
+    const allSchemaSaids = Object.keys(message.exn.e.acdc?.e || {}).map(
+      // Chained schemas
+      (key) => message.exn.e.acdc.e?.[key]?.s
+    );
+    allSchemaSaids.push(schemaSaid);
+    await Promise.all(
+      allSchemaSaids.map(
+        async (schemaSaid) =>
+          await Agent.agent.connections.resolveOobi(
+            `${ConfigurationService.env.keri.credentials.testServer.urlInt}/oobi/${schemaSaid}`,
+            true
+          )
+      )
+    );
+
+    const schema = await this.props.signifyClient
+      .schemas()
+      .get(schemaSaid)
+      .catch((error) => {
+        const errorStack = (error as Error).stack as string;
+        const status = errorStack.split("-")[1];
+        if (/404/gi.test(status) && /SignifyClient/gi.test(errorStack)) {
+          return undefined;
+        } else {
+          throw error;
+        }
+      });
+
     await this.ipexMessageStorage.createIpexMessageRecord({
```

Comment:
I also asked to store the schema SAID as a new field in the IPEX message record called `schema`. This is so if we have to store `undefined` we can fix it later.

Reply:
We can get the schema SAID in the stored `content` so I don't think we need to store it.

Reply:
Actually, good point, we can always retrieve it from there or KERIA.

But I think we've another issue here. `exn.e.acdc` is attached to `grant` and `offer` messages.

It's not attached to `apply`, `agree`, `admit` messages.

In this function can you handle the different cases (and test):
- `grant` is as you have it `message.exn.e.acdc.s`
- `apply` is `message.exn.a.s`
- For `agree`, the schema isn't included but `message.exn.p` points to an `offer` message - so I think we can call `(await this.props.signifyClient.exchanges().get(message.exn.p)).exn.e.acdc.s` or something like that